esphome/components/shelly_dimmer/shelly_dimmer.cpp

```diff
@@ -0,0 +1,490 @@
+#include "esphome/core/defines.h"
+#include "shelly_dimmer.h"
+#include "stm32flash.h"
+#ifndef USE_ESP_IDF
+#include <HardwareSerial.h>
+#endif
+
+#include <algorithm>
+#include <cstring>
+#include <memory>
+#include <numeric>
+
+namespace {
+
+constexpr char TAG[] = "shelly";
```

Comment:
For consistency with codebase
```suggestion
static const char *const TAG = "shelly_dimmer";
```

Reply:
Makes more sense to have constexpr as it is a compile time constant. By changing the type from `char[]` to `char*` you'll loose the size known at compile time.

Reply:
I am fine with both! :)